test/gateway_composer/transfer_token.js

```diff
@@ -0,0 +1,134 @@
+// Copyright 2018 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const BN = require('bn.js');
+const utils = require('../test_lib/utils');
+const { AccountProvider } = require('../test_lib/utils.js');
+
+const gatewayComposerUtils = require('./utils');
+
+contract('GatewayComposer::transferToken', async (accounts) => {
+    describe('Negative Tests', async () => {
+        const accountProvider = new AccountProvider(accounts);
+
+        it('Should Fail when owner is not the caller.', async () => {
+            const {
+                gatewayComposer,
+                valueToken,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const to = accountProvider.get();
+            const amount = new BN(10);
+            utils.expectRevert(gatewayComposer.transferToken(
+                valueToken.address,
+                to,
+                amount,
+                { from: accountProvider.get() },
+            ),
+            'Should revert as msg.sender is not the owner.',
+            'Only owner can call the function.');
+        });
+
+        it('Should fail when EIP20 token address is invalid.', async () => {
+            const {
+                gatewayComposer,
+                owner,
+            } = await gatewayComposerUtils.setupGatewayComposer(accountProvider);
+
+            const to = accountProvider.get();
+            const amount = new BN(10);
+            utils.expectRevert(gatewayComposer.transferToken(
+                utils.NULL_ADDRESS,
+                to,
+                amount,
+                { from: owner },
+            ),
+            'Should revert as token address is null.',
+            'EIP20 token address is null.');
+        });
+
+        it('Should fail when owner balance is not sufficient.', async () => {
```

Comment:
Suggest revising the `it` here. Like the tests that depend on Gateway or BT returning `false` but do not go into why, this test is really just a test of what to do when interacting with an arbitrary address returns false.

Reply:
Updated it to : "Fails when ValueToken transfer returned false."